### PR TITLE
Post date time feature

### PR DESCRIPTION
**Pull request by j311y**
Main Changes

- added `config.time_zone = "London"` to all config/environments
- passed `Time.now` as parameter to `posts` when created
- format time method in `models/post.rb`
- add formatted time to `posts Timeline`

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,2 +1,5 @@
 class Post < ApplicationRecord
+  def format_time
+    return "posted at #{self.created_at.strftime('%H:%M')} on #{self.created_at.strftime('%d/%m/%Y')}"
```
Comment (dbugsy) [inline app/models/post.rb]: This shouldn't be in the model - it is presentation logic. Prefer a PostPresenter object instead.

File: spec/features/user_can_submit_posts_spec.rb [hunk 2]
```diff
@@ -8,4 +8,12 @@
...
     expect(page).to have_content("Hello, world!")
   end
+
+  scenario "Posts display with date and time" do
+    visit "/posts"
+    click_link "New post"
+    fill_in "Message", with: "Hello, world!"
+    click_button "Submit"
+    expect(page).to have_content("posted at #{Time.now.strftime('%H:%M')} on #{Time.now.strftime('%d/%m/%Y')}")
+  end
```
Comment (dbugsy) [inline spec/features/user_can_submit_posts_spec.rb]: Duplicates the test above. Refactor to remove duplication.

**Review by dbugsy — CHANGES_REQUESTED** [review]

**Comment by j311y** [comment]
Requested changes made - PostPresenter module added

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -15,6 +15,6 @@ def index
   private
 
   def post_params
-    params.require(:post).permit(:message)
+    params.require(:post).permit(:message, created_at: Time.now)
```
Comment (dbugsy) [inline app/controllers/posts_controller.rb]: I don't think you need to add this to the permit, as the timestamp will be added at the model layer.

File: app/models/post_presenter.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+module PostPresenter
+
+  def format_time
+    return "posted at #{self.created_at.strftime('%H:%M')} on #{self.created_at.strftime('%d/%m/%Y')}"
+  end
+end
```
Comment (dbugsy) [inline app/models/post_presenter.rb]: Looks great! I'd probably put it into 'app/presenters/post_presenter.rb'.

A class is more commonly used:

```ruby
class PostPresenter
  def initialize(post)
    @post = post
  end
  
  def timestamp
    post.created_at.strftimeblablablabl
  end
end

```

Comment (dbugsy) [reply app/models/post_presenter.rb]: The "posted at etc" is purely view based, and should appear in the view template if at all possible.

File: app/models/post_presenter.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+module PostPresenter
+
+  def format_time
+    return "posted at #{self.created_at.strftime('%H:%M')} on #{self.created_at.strftime('%d/%m/%Y')}"
```
Comment (dbugsy) [inline app/models/post_presenter.rb]: implicit return in ruby

**Review by dbugsy — CHANGES_REQUESTED** [review]

**Comment by j311y** [comment]
Requested changes made

**Comment by j311y** [comment]
Also don't understand why Travis CI build is failing. Any ideas?

UPDATE

Travis CI issue fixed

**Comment by axcochrane** [comment]
Hey - as we just chatted about. Could potentially make separate methods for datestamp and timestamp so they can be reused elsewhere if needed.

```
class PostPresenter
  def initialize(post)
    @post = post
  end
  
  def timestamp
    post.created_at.strftime('%H:%M')
  end

  def datestamp
    post.created_at.strftime('%d/%m/%Y')
  end
end
```

**Review by axcochrane — CHANGES_REQUESTED** [review]
Hey - as we just chatted about. Could potentially make separate methods for datestamp and timestamp so they can be reused elsewhere if needed.
```
class PostPresenter
  def initialize(post)
    @post = post
  end
  
  def timestamp
    post.created_at.strftime('%H:%M')
  end

  def datestamp
    post.created_at.strftime('%d/%m/%Y')
  end
end
```

**Comment by j311y** [comment]
Display logic separated to /views - still using a module over a class as cannot extract methods from class.

**Review by dbugsy — APPROVED** [review]

**Comment by dbugsy** [comment]
You would just use the presenter class in the controller instead of the Post class, but if the module pattern works for you, I'm happy to try it.

**Review by anaalta — APPROVED** [review]

**Comment by adc17** [comment]
Wicked guys :shipit:

**Comment by anaalta** [comment]
:shipit:  :) nice

**Review by AlexChalk — APPROVED** [review]
Nice :shipit:

**Pull request merged**